### PR TITLE
_extras/guide.md: update to match episodes' contents

### PR DESCRIPTION
**Pull request by rhaas80**
changes:
* site_id -> plot_id
* fixed default number of lines for head() and tail()
* misc. stuff

Please delete the text below before submitting your contribution.

### CODE REVIEW

**Comment by maxim-belkin** [comment]
> You may want to reword the "Please delete the text below before submitting your contribution." instruction to "Please delete this line and the text below before submitting your contribution." to avoid mistakes like the one I did just now above.

This text is common across Carpentries' repositories and the change has to happen in the caprentries/styles repo. The file is `.github/PULL_REQUEST_TEMPLATE.md`.

File: _extras/guide.md [hunk 1]
```diff
@@ -225,17 +225,17 @@ Tip: use `.head()` method throughout this lesson to keep your display neater for
...
+	For example, using `PB` and `PL`:  `surveys_df[surveys_df['species_id'].isin(['PB', 'PL'])]['plot_id'].unique()` provides a list of the plots with these species involved. With `surveys_df[surveys_df['species_id'].isin(['PB', 'PL'])].shape` the number of records can be derived.
 
 * Create a query that finds all rows with a weight value > or equal to 0.
 
 	`surveys_df[surveys_df["weight"] >= 0]`
 
 	*Suggestion*: Introduce already that all these slice operations are actually based on a *Boolean indexing* operation (next section in the lesson). The filter provides for each record if it satisfies (True) or not (False). The slicing itself interprets the True/False of each record.
 
-* The `~` symbol in Python can be used to return the OPPOSITE of the selection that you specify in Python. It is equivalent to is not in. Write a query that selects all rows that are NOT equal to 'M' or 'F' in the surveys data.
+* The `~` symbol in Python can be used to return the OPPOplot of the selection that you specify in Python. It is equivalent to is not in. Write a query that selects all rows that are NOT equal to 'M' or 'F' in the surveys data.
```
Comment (maxim-belkin) [inline _extras/guide.md]: ahem :)

File: _extras/guide.md [hunk 2]
```diff
@@ -332,65 +332,65 @@ merged_left = pd.merge(left=surveys_df,right=species_df, how='left', on="species
...
 ~~~
-merged_left.groupby(["site_id", "taxa"]).count()["record_id"].unstack().plot(kind='bar', stacked=True)
+merged_left.groupby(["plot_id", "taxa"]).count()["record_id"].unstack().plot(kind='bar', stacked=True)
 plt.legend(loc='upper center', ncol=3, bbox_to_anchor=(0.5, 1.05))
 ~~~
 {: .language-python}
 (the legend otherwise overlaps the bar plot)
 
-![taxa per site](../fig/04_chall_taxa_per_site.png)
+![taxa per plot](../fig/04_chall_taxa_per_plot.png)
```
Comment (maxim-belkin) [inline _extras/guide.md]: image is still `_per_site`...

File: _extras/guide.md [hunk 3]
```diff
@@ -332,65 +332,65 @@ merged_left = pd.merge(left=surveys_df,right=species_df, how='left', on="species
...
-Species distribution (number of taxa for each site) can be derived as follows:
+Species distribution (number of taxa for each plot) can be derived as follows:
 ~~~
-merged_left.groupby(["site_id"])["taxa"].nunique().plot(kind='bar')
+merged_left.groupby(["plot_id"])["taxa"].nunique().plot(kind='bar')
 ~~~
 {: .language-python}
 
-![taxa per site](../fig/04_chall_ntaxa_per_site.png)
+![taxa per plot](../fig/04_chall_ntaxa_per_plot.png)
```
Comment (maxim-belkin) [inline _extras/guide.md]: image name is still the same (`_per_site`)

File: _extras/guide.md [hunk 4]
```diff
@@ -332,65 +332,65 @@ merged_left = pd.merge(left=surveys_df,right=species_df, how='left', on="species
...
+ntaxa_sex_plot= merged_left.groupby(["plot_id", "sex"])["taxa"].nunique().reset_index(level=1)
+ntaxa_sex_plot = ntaxa_sex_plot.pivot_table(values="taxa", columns="sex", index=ntaxa_sex_plot.index)
+ntaxa_sex_plot.plot(kind="bar", legend=False)
 plt.legend(loc='upper center', ncol=3, bbox_to_anchor=(0.5, 1.08),
            fontsize='small', frameon=False)
 ~~~
 {: .language-python}
 
-![taxa per site per sex](../fig/04_chall_ntaxa_per_site_sex.png)
+![taxa per plot per sex](../fig/04_chall_ntaxa_per_plot_sex.png)
```
Comment (maxim-belkin) [inline _extras/guide.md]: figure name is still the same

File: _extras/guide.md [hunk 5]
```diff
@@ -332,65 +332,65 @@ merged_left = pd.merge(left=surveys_df,right=species_df, how='left', on="species
...
-sex_taxa_site.unstack(level=[1, 2]).plot(kind='bar', logy=True)
+sex_taxa_plot  = merged_left.groupby(["plot_id", "taxa", "sex"]).count()['record_id']
+sex_taxa_plot.unstack(level=[1, 2]).plot(kind='bar', logy=True)
 plt.legend(loc='upper center', ncol=3, bbox_to_anchor=(0.5, 1.15),
            fontsize='small', frameon=False)
 ~~~
 {: .language-python}
 
-![taxa per site per sex](../fig/04_chall_sex_taxa_site_intro.png)
+![taxa per plot per sex](../fig/04_chall_sex_taxa_plot_intro.png)
```
Comment (maxim-belkin) [inline _extras/guide.md]: figure name is still the same

File: _extras/guide.md [hunk 6]
```diff
@@ -400,28 +400,28 @@ axs[0].legend(loc='upper center', ncol=5, bbox_to_anchor=(0.5, 1.3),
 ~~~
 {: .language-python}
 
-![taxa per site per sex](../fig/04_chall_sex_taxa_site.png)
+![taxa per plot per sex](../fig/04_chall_sex_taxa_plot.png)
```
Comment (maxim-belkin) [inline _extras/guide.md]: same comment as before

File: _extras/guide.md [hunk 7]
```diff
@@ -435,10 +435,10 @@ plt.xlabel("Diversity index")
 ~~~
 {: .language-python}
 
-![taxa per site per sex](../fig/04_chall_diversity_index.png)
+![taxa per plot per sex](../fig/04_chall_diversity_index.png)
```
Comment (maxim-belkin) [inline _extras/guide.md]: same comment as before

File: _extras/guide.md [hunk 8]
```diff
@@ -694,6 +698,7 @@ plt.show()
 
 
 
+
```
Comment (maxim-belkin) [inline _extras/guide.md]: Let's keep just one new line at the end of the file.

File: _extras/guide.md [hunk 9]
```diff
@@ -640,20 +640,20 @@ def yearly_data_csv_writer(all_data, yearcolumn="year",
...
 
-## 07-putting-it-all-together
+## 08-putting-it-all-together
 
 Scientists often operate on mathematical equations. Being able to use them in their graphics has a
 lot of added value. Luckily, Matplotlib provides powerful tools for text control. One of them is the
 ability to use LaTeX mathematical notation, whenever text is used (you can learn more about LaTeX
 math notation here: https://en.wikibooks.org/wiki/LaTeX/Mathematics). To use mathematical notation,
-surround your text using the dollar sign ("$").  LaTeX uses the backslash character ("\") a lot.
+surround your text using the dollar sign ("$").  LaTeX uses the backslash character ("\\") a lot.
```
Comment (maxim-belkin) [inline _extras/guide.md]: 👍

File: _extras/guide.md (hunk 1 above)
Comment (wrightaprilm) [reply _extras/guide.md]: Oh, that's funny.

**Comment by wrightaprilm** [comment]
Thanks for this, Roland.  I think this should be good to go once you've made the changes requested by @maxim-belkin (thanks for looking at this, as well @maxim-belkin).

File: _extras/guide.md (hunk 1 above)
Comment (rhaas80) [reply _extras/guide.md]: a search an replace foobar. Let me fix this...

File: _extras/guide.md (hunk 8 above)
Comment (rhaas80) [reply _extras/guide.md]: There were literally 26 empty lines at the end. I have reduced this to one now.

**Comment by rhaas80** [comment]
I think I took care of the requested changes. I assume this will be squashed when merged?

**Comment by wrightaprilm** [comment]
Sorry I'm so late in getting back to this - coming off maternity leave this week. It looks like all the changes are in, and we're good. Thanks so much for this, @rhaas80.

**Pull request merged**